Plugins/BuildServerIntegration/JenkinsIntegration/JenkinsAdapter.cs

```diff
@@ -320,6 +324,36 @@ private void ObserveBuilds(DateTime? sinceDate, bool? running, IObserver<BuildIn
                     observer.OnError(ex);
                 }
             }
+
+            return;
+
+            // Sort order: Completed > InProgress > Aborted/Stopped > None
+            static bool StatusIsBetter(BuildInfo.BuildStatus newStatus, ObjectId commit, Dictionary<ObjectId, BuildInfo.BuildStatus> builds)
+            {
+                return
+
+                    // No existing status
+                    !builds.ContainsKey(commit)
+
+                    // Completed status is never replaced
+                    || (!IsBuildCompleted(builds[commit])
+
+                        // Existing status is lower than Completed
+                        && (IsBuildCompleted(newStatus)
+
+                           || (builds[commit] != BuildInfo.BuildStatus.InProgress
+
+                               // Existing status is lower than InProgress
+                               && newStatus != BuildInfo.BuildStatus.Unknown
+                                   && newStatus != BuildInfo.BuildStatus.Stopped)));
```

Comment:
I can tell you I'm definitely not a fan of this. It is almost impossible to see conditions boundaries. Can we restructure this with the use of temp variables?

Reply:
I rewrote using serial if statements instead